src/core.h

```diff
@@ -442,3 +477,31 @@ struct media_track *mediatrack_lookup_media(const struct list *medial,
 					    struct stream *strm);
 void mediatrack_close(struct media_track *media, int err);
 void mediatrack_sdp_attr_decode(struct media_track *media);
+
+/*
+ * Stream receiver
+ */
+int rx_alloc(struct receiver **rxp,
+	     struct stream *strm,
+	     const char *name,
+	     const struct config_avt *cfg,
+	     stream_rtp_h *rtph,
+	     stream_pt_h *pth, void *arg);
+void rx_set_handlers(struct receiver *rx,
+		     stream_rtpestab_h *rtpestabh, void *arg);
+struct metric *rx_metric(struct receiver *rx);
+struct jbuf *rx_jbuf(struct receiver *rx);
+void rx_receive(const struct sa *src, const struct rtp_header *hdr,
+	       struct mbuf *mb, void *arg);
+void rx_handle_rtcp(const struct sa *src, struct rtcp_msg *msg, void *arg);
+void rx_set_ssrc(struct receiver *rx, uint32_t ssrc);
+uint64_t rx_ts_last(struct receiver *rx);
+void rx_set_ts_last(struct receiver *rx, uint64_t ts_last);
+void rx_flush(struct receiver *rx);
+void rx_set_enable(struct receiver *rx, bool enable);
+int rx_get_ssrc(struct receiver *rx, uint32_t *ssrc);
+int rx_debug(struct re_printf *pf, const struct receiver *rx);
+int rx_start_thread(struct receiver *rx, struct rtp_sock *rtp);
+void rx_mnat_connected_handler(const struct sa *raddr1,
+			       const struct sa *raddr2, void *arg);
+bool rx_running(const struct receiver *rx);
```

Comment:
perhaps "rx_" prefix is a bit short for the global namespace.

how about `rtp_receiver` or something like that ?


We should consider borrowing API names from the WebRTC JS api:
https://developer.mozilla.org/en-US/docs/Web/API/RTCRtpReceiver